cmd/catchpointdump/file.go

```diff
@@ -380,7 +386,64 @@ func printAccountsDatabase(databaseName string, fileHeader ledger.CatchpointFile
 		}
 
 		// increase the deadline warning to disable the warning message.
-		db.ResetTransactionWarnDeadline(ctx, tx, time.Now().Add(5*time.Second))
+		_, _ = db.ResetTransactionWarnDeadline(ctx, tx, time.Now().Add(5*time.Second))
 		return err
 	})
 }
+
+func printKeyValue(writer *bufio.Writer, key, value []byte) {
+	var pretty string
+	ai, rest, err := logic.SplitBoxKey(string(key))
+	if err == nil {
+		pretty = fmt.Sprintf("box(%d, %s)", ai, base64.StdEncoding.EncodeToString([]byte(rest)))
+	} else {
+		pretty = base64.StdEncoding.EncodeToString(key)
+	}
+
+	fmt.Fprintf(writer, "%s : %v\n", pretty, base64.StdEncoding.EncodeToString(value))
+}
+
+func printKeyValueStore(databaseName string, outFile *os.File) error {
+	fmt.Printf("\n")
+	printDumpingCatchpointProgressLine(0, 50, 0)
+	lastProgressUpdate := time.Now()
+	progress := uint64(0)
+	defer printDumpingCatchpointProgressLine(0, 0, 0)
+
+	fileWriter := bufio.NewWriterSize(outFile, 1024*1024)
+	defer fileWriter.Flush()
+
+	dbAccessor, err := db.MakeAccessor(databaseName, true, false)
+	if err != nil || dbAccessor.Handle == nil {
+		return err
+	}
+
+	return dbAccessor.Atomic(func(ctx context.Context, tx *sql.Tx) error {
+		var rowsCount int64
+		err := tx.QueryRow("SELECT count(*) from catchpointkvstore").Scan(&rowsCount)
+		if err != nil {
+			return err
+		}
+
+		rows, err := tx.Query("SELECT key, value FROM catchpointkvstore")
```

Comment:
Is this only so that the catchpoint dump utility prints in the same order?  I want to check my understanding that the catchpoint files don't actually need to be identical (down to the level of ordering) for any other reason. All that matters for their use that they produce the same merkle trie, which is order independent. (I think.)

I'll certainly make the change, but I want to be sure I know what's going on.

Reply:
Oh! This is actually in the catchpoint dump utility! So certainly we're talking about the same reason.

So now, just to be extra safe, can you confirm that the similar SELECT, in `catchpointwriter.go` does NOT need to be ordered?

```
	// Create the *Rows iterator JIT
	if cw.kvRows == nil {
		rows, err := tx.QueryContext(ctx, "SELECT key, value FROM kvstore")
		if err != nil {
			return err
		}
		cw.kvRows = rows
	}
```